src/components/view/view-event-list.tsx

```diff
@@ -879,19 +881,37 @@ export class ViewEventList extends React.Component<ViewEventListProps> {
         if (!listWindow) return true; // This means no rows, so we are effectively at the bottom
         else return (listWindow.scrollTop + SCROLL_BOTTOM_MARGIN) >= (listWindow.scrollHeight - listWindow.offsetHeight);
     }
-
     private wasListAtBottom = true;
     private updateScrolledState = () => {
         requestAnimationFrame(() => { // Measure async, once the scroll has actually happened
             this.wasListAtBottom = this.isListAtBottom();
+            
+            // Only save scroll position after we've restored the initial state
+            if (this.hasRestoredInitialState) {
+                const listWindow = this.listBodyRef.current?.parentElement;
+                if (listWindow) {
+                    this.props.uiStore.setViewScrollPosition(listWindow.scrollTop);
+                }
+            }
         });
     }
 
+    private hasRestoredInitialState = false;    
     componentDidMount() {
-        this.updateScrolledState();
+        // Don't save scroll state immediately - wait until we've restored first
+        
+        // Use a more aggressive delay to ensure DOM is fully ready
+        setTimeout(() => {
+            this.restoreScrollPosition();
+            
+            // Only start tracking scroll changes after we've restored
+            setTimeout(() => {
+                this.hasRestoredInitialState = true;
+            }, 100);
+        }, 100);
```

Comment:
Would be good to avoid the fixed timeouts here. I think we can do so though - we just need to wait until listBodyRef is ready, right? I think we can convert this into a callback ref, which means we can trigger the DOM interaction (updating the scroll position) immediately after that's rendered.

That will be a bit more responsive (no visible jumping around I hope) and should avoid most of the risk of the race condition you're protecting against with `hasRestoredInitialState` here as well. Does that make sense?